File: dialogs/src/main/java/com/rands/lightbulb/dialogs/dialog_loading/LoadingDialog.java
package com.rands.lightbulb.dialogs.dialog_loading;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;

import com.rands.lightbulb.dialogs.R;
import com.rands.lightbulb.dialogs.base.BaseDialogFragment;

public class LoadingDialog extends BaseDialogFragment {

    public static LoadingDialog newInstance(String title, String message, DialogTypes dialogType, DialogAnimationTypes animationType) {
        LoadingDialog dialogFragment = new LoadingDialog();
        dialogFragment.setArguments(new DialogBundleHelper()
                .withTitle(title)
                .withMessage(message)
                .withCancelable(false)
                .withShowing(false)
                .withDialogType(dialogType == null ? DialogTypes.NORMAL : dialogType)
                .withAnimation(animationType == null ? DialogAnimationTypes.NO_ANIMATION : animationType)
                .getBundle()
        );
        return dialogFragment;
    }

    @Override
    protected View setDialogLayout(LayoutInflater inflater) {
        return View.inflate(getContext(), R.layout.dialog_loading_normal, null);
    }

    @Override
    protected void configureContent(View view, Bundle savedInstanceState) {
    }
}

File: lightbulb-dialogs/src/main/java/com/github/RooneyAndShadows/lightbulb/dialogs/dialog_loading/LoadingDialog.kt
package com.github.rooneyandshadows.lightbulb.dialogs.dialog_loading

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import com.github.rooneyandshadows.lightbulb.dialogs.base.BaseDialogFragment
import com.github.rooneyandshadows.lightbulb.dialogs.R
import com.github.rooneyandshadows.lightbulb.dialogs.base.internal.DialogAnimationTypes
import com.github.rooneyandshadows.lightbulb.dialogs.base.internal.DialogBundleHelper
import com.github.rooneyandshadows.lightbulb.dialogs.base.internal.DialogTypes

class LoadingDialog : BaseDialogFragment() {
    @Override
    override fun getDialogLayout(layoutInflater: LayoutInflater): View {
        return View.inflate(context, R.layout.dialog_loading_normal, null)
    }

    @Override
    override fun configureContent(view: View, savedInstanceState: Bundle?) {
    }

    companion object {
        fun newInstance(
            title: String?,
            message: String?,
            dialogType: DialogTypes = DialogTypes.NORMAL,
            animationType: DialogAnimationTypes = DialogAnimationTypes.NO_ANIMATION
        ): LoadingDialog {
            val dialogFragment = LoadingDialog()
            dialogFragment.arguments = DialogBundleHelper()
                .withTitle(title)
                .withMessage(message)
                .withCancelable(false)
                .withShowing(false)
                .withDialogType(dialogType)
                .withAnimation(animationType)
                .bundle
            return dialogFragment
        }
    }
}